hubploy/auth.py

```diff
@@ -43,6 +47,39 @@ def registry_auth_gcloud(deployment, project, service_key):
     ])
 
 
+def registry_auth_aws(deployment, project, service_key):
+    """
+    Setup AWS authentication with a service_key
+
+    This changes *global machine state* on where docker can push to!
+    """
+    import boto3
+    
+    service_key_path = os.path.join(
+        'deployments', deployment, 'secrets', service_key
+    )
+
+    if not os.path.isfile(service_key_path):
+        raise FileNotFoundError(f'The service_key file {service_key_path} does not exist')
+
+    os.environ['AWS_SHARED_CREDENTIALS_FILE'] = os.path.abspath(service_key_path)
+
+    client = boto3.client(
+        'ecr'
+    )
+    response = client.get_authorization_token()
+    auth_dict = response['authorizationData'][0]
+
+    authorizationToken = auth_dict['authorizationToken']
+    proxyEndpoint = auth_dict['proxyEndpoint']
+
+    subprocess.check_call([
```

Comment:
This expects the docker commandline client to be installed, which we don't. For a 'quick' fix you can install it, but the right thing to do is either to create the ~/.docker/config.json file manually ourselves in this code, or use https://github.com/awslabs/amazon-ecr-credential-helper. The latter is the recommended method - we use the equivalent with Google Cloud.

Reply:
amazon-ecr-credential-helper now installed in .circleci/config.yml and the docker config.json is written in registry_auth_aws(). is that all that needs to happen in registry_auth_aws? it seems like everything is authenticated so long as this environment variable is present: `AWS_SHARED_CREDENTIALS_FILE`